apps/prairielearn/src/pages/administratorQuery/administratorQuery.ts

```diff
@@ -0,0 +1,140 @@
+import express = require('express');
+const router = express.Router();
+import asyncHandler = require('express-async-handler');
+import fsPromises = require('node:fs/promises');
+import path = require('path');
+import hljs from 'highlight.js';
+import { stringify } from '@prairielearn/csv';
+import { z } from 'zod';
+
+import jsonLoad = require('../../lib/json-load');
+import sqldb = require('@prairielearn/postgres');
+import {
+  AdministratorQuery,
+  AdministratorQuerySchema,
+  AdministratorQueryQueryRunSchema,
+  AdministratorQueryRunParams,
+  AdministratorQueryQueryRun,
+} from './administratorQuery.html';
+
+const sql = sqldb.loadSqlEquiv(__filename);
+
+const queriesDir = path.resolve(__dirname, '..', '..', 'admin_queries');
+
+const AdministratorQueryQueryRunIDSchema = z.string();
+
+router.get(
+  '/:query',
+  asyncHandler(async (req, res, next) => {
+    const jsonFilename = req.params.query + '.json';
+    const sqlFilename = req.params.query + '.sql';
+
+    const info = AdministratorQuerySchema.parse(
+      await jsonLoad.readJSONAsync(path.join(queriesDir, jsonFilename)),
+    );
+    const querySql = await fsPromises.readFile(path.join(queriesDir, sqlFilename), {
+      encoding: 'utf8',
+    });
+    const sqlHighlighted = hljs.highlight(querySql, {
+      language: 'sql',
+    }).value;
+
+    let has_query_run = false;
```

Comment:
I think in practice we don't need this variable; now we can just rely on if `query_run` is null or not. That is, we can replace `if (!has_query_run)` with `if (!query_run)`. This simplifies things a little bit and should make TypeScript work a little better for us. Specifically, if we use `if (query_run)`, TypeScript will know that `query_run` isn't `null` inside that conditional so we won't have to use optional chaining to access properties. Usinga separate `has_query_run` boolean, TypeScript can't connect that variable to the fact that `query_run` is null or not.

Reply:
That makes sense. I have removed `has_query_run` and replaced all usage with `if(query_run)` in commit 88bbb1f.